File: 2-Medium/Stock Exchange Losses/stock-exchange-losses.js
function calculateMaxLoss(values, n) {
    let maxLoss = 0;
    let currentMax = values[0];

    for (let i = 1; i < n; i++) {
        if (values[i] > currentMax) {
            currentMax = values[i];
        }
        else {
            maxLoss = Math.max(maxLoss, currentMax - values[i]);
        }
    }

    return -maxLoss || 0;
}

const n = parseInt(readline());
const inputs = readline().split(' ').map(Number);

const maxLoss = calculateMaxLoss(inputs, n);
console.log(maxLoss);

File: 2-Medium/Stock Exchange Losses/stock-exchange-losses.cs
using System;

class Solution
{
    static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());
        int[] values = Array.ConvertAll(Console.ReadLine().Split(' '), int.Parse);

        int maxLoss = CalculateMaxLoss(values, n);
        Console.WriteLine(maxLoss);
    }

    static int CalculateMaxLoss(int[] values, int n)
    {
        int maxLoss = 0;
        int currentMax = values[0];

        for (int i = 1; i < n; i++)
        {
            if (values[i] > currentMax)
            {
                currentMax = values[i];
            }
            else
            {
                maxLoss = Math.Max(maxLoss, currentMax - values[i]);
            }
        }

        return -maxLoss;
    }
}
